meta-resin-common/recipes-support/hostapp-update-hooks/files/99-resin-grub

```diff
@@ -15,11 +15,11 @@ else
 fi
 
 new_part=$(findmnt --noheadings --canonicalize --output SOURCE $SYSROOT)
-new_part_label=$(blkid "$new_part" | awk '{print $2}')
+new_part_label=$(blkid "$new_part" | awk '{print $2}' | cut -d'"' -f 2)
 
 printf "[INFO] Switching root partition to %s..." "$new_part_label..."
 grub_cfg=$(find /mnt/boot/ -name grub.cfg)
-sed "s/root=[^ ]* /"root=$new_part_label" /" "$grub_cfg" > "$grub_cfg".new
+sed "s/resin-root./"$new_part_label"/" "$grub_cfg" > "$grub_cfg".new
 
```

Comment:
add a transition hook to remove kernel from boot partition and add extra lines in grub.cfg